motionMonitor/motionMonitor.c

```diff
@@ -1,116 +1,158 @@
- */
-le_result_t brnkl_motion_getCurrentAcceleration(double* xAcc,
-                                                double* yAcc,
-                                                double* zAcc) {
-  le_result_t r;
-  char path[256];
-
-  double scaling = 0.0;
-  int pathLen = snprintf(path, sizeof(path), FormatStr, AccType, CompScale);
-  LE_ASSERT(pathLen < sizeof(path));
-  r = ioutil_readDoubleFromFile(path, &scaling);
-  if (r != LE_OK) {
-    goto done;
-  }
+*/
 
-  pathLen = snprintf(path, sizeof(path), FormatStr, AccType, CompX);
-  LE_ASSERT(pathLen < sizeof(path));
-  r = ioutil_readDoubleFromFile(path, xAcc);
-  if (r != LE_OK) {
-    goto done;
-  }
-  *xAcc *= scaling;
+le_result_t brnkl_motion_getCurrentAcceleration(
+    double *xAcc,
+    double *yAcc,
+    double *zAcc
+)
+{
+    le_result_t r;
+    char path[256];
 
-  pathLen = snprintf(path, sizeof(path), FormatStr, AccType, CompY);
-  LE_ASSERT(pathLen < sizeof(path));
-  r = ioutil_readDoubleFromFile(path, yAcc);
-  if (r != LE_OK) {
-    goto done;
-  }
-  *yAcc *= scaling;
+    double scaling = 0.0;
+    r = ioutil_readDoubleFromFile(path, &scaling);
+    if (r != LE_OK)
+    {
+        goto done;
+    }
+
+    r = ioutil_readDoubleFromFile(path, xAcc);
+    if (r != LE_OK)
+    { 
+        goto done;
+    }
+    *xAcc *= scaling;
+
+    r = ioutil_readDoubleFromFile(path, yAcc);
+    if (r != LE_OK)
+    {
+        goto done;
+    }
+    *yAcc *= scaling;
 
-  pathLen = snprintf(path, sizeof(path), FormatStr, AccType, CompZ);
-  LE_ASSERT(pathLen < sizeof(path));
-  r = ioutil_readDoubleFromFile(path, zAcc);
-  *zAcc *= scaling;
+    r = ioutil_readDoubleFromFile(path, zAcc);
+    *zAcc *= scaling;
 
 done:
-  LE_INFO("Showing accel: x: %f, y: %f, z: %f", *xAcc, *yAcc, *zAcc);
-  return r;
+    LE_INFO("Showing accel X: %f Y: %f Z: %f ", *xAcc, *yAcc, *zAcc);
+    return r;
 }
 
-le_result_t brnkl_motion_getSuddenImpact(double* xAcc,
-                                         double* yAcc,
-                                         double* zAcc) {
-  *xAcc = suddenImpact.x;
-  *yAcc = suddenImpact.y;
-  *zAcc = suddenImpact.z;
-  suddenImpact.x = 0;
-  suddenImpact.y = 0;
-  suddenImpact.z = 0;
-  hasSuddenImpact = false;
-  return LE_OK;
+void recordImpact(double* xAcc, double* yAcc, double* zAcc){
+  timestamps[totalImpacts] = GetCurrentTimestamp();
+  xAccImpact[totalImpacts] = *xAcc;
+  yAccImpact[totalImpacts] = *yAcc;
+  zAccImpact[totalImpacts] = *zAcc;
+  totalImpacts++;
+
+  LE_INFO("New Impact, totalImpacts: %d", totalImpacts);
 }
 
-int8_t brnkl_motion_hasSuddenImpact() {
-  return hasSuddenImpact;
+le_result_t brnkl_motion_getSuddenImpact(double* xAcc, size_t *xSize,
+              double* yAcc, size_t *ySize,
+              double* zAcc, size_t *zSize) {
+
+  if(!totalImpacts)
+    LE_INFO("No Sudden Impacts to Report");
+
+  //check 
+  if( 
+    totalImpacts > sizeof(xAccImpact) ||
```

Comment:
I would suggest moving this to where readings get recorded (`recordImpact`). It may be worth checking `sizeof(xAccImpact) < (*xSize)` here though.